http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/RuntimeServiceInterp.scala

```diff
@@ -0,0 +1,316 @@
+      _ <- if (hasPermission) F.unit else F.raiseError[Unit](AuthorizationError(Some(userInfo.userEmail)))
+      // Grab the service accounts from serviceAccountProvider for use later
+      clusterServiceAccountOpt <- serviceAccountProvider
+        .getClusterServiceAccount(userInfo, googleProject)
+      notebookServiceAccountOpt <- serviceAccountProvider
+        .getNotebookServiceAccount(userInfo, googleProject)
+      serviceAccountInfo = ServiceAccountInfo(clusterServiceAccountOpt, notebookServiceAccountOpt)
+
+      clusterOpt <- clusterQuery.getActiveClusterByNameMinimal(googleProject, runtimeName).transaction
+
+      _ <- clusterOpt match {
+        case Some(c) => F.raiseError[Unit](RuntimeAlreadyExistsException(googleProject, runtimeName, c.status))
+        case None =>
+          for {
+            internalId <- F.pure(RuntimeInternalId(context.traceId.asString)) //TODO: is there any downside to reusing traceId as internalId?
+            petToken <- serviceAccountProvider.getAccessToken(userInfo.userEmail, googleProject).recoverWith {
+              case e =>
+                log.warn(e)(
+                  s"Could not acquire pet service account access token for user ${userInfo.userEmail.value} in project $googleProject. " +
+                    s"Skipping validation of bucket objects in the cluster request."
+                ) as None
+            }
+            clusterImages <- getClusterImages(petToken,
+                                              userInfo.userEmail,
+                                              googleProject,
+                                              context.now,
+                                              req.toolDockerImage,
+                                              req.welderDockerImage)
+            cluster <- F.fromEither(
+              convertToRuntime(userInfo,
+                               serviceAccountInfo,
+                               googleProject,
+                               runtimeName,
+                               internalId,
+                               clusterImages,
+                               config,
+                               req)
+            )
+            gcsObjectUrisToValidate = cluster.userJupyterExtensionConfig
+              .map(
+                config =>
+                  (config.nbExtensions.values ++ config.serverExtensions.values ++ config.combinedExtensions.values)
+                    .filter(_.startsWith("gs://"))
+                    .toList
+              )
+              .getOrElse(List.empty) ++ req.jupyterUserScriptUri.map(_.asString) ++ req.jupyterStartUserScriptUri.map(
+              _.asString
+            )
+            _ <- petToken.traverse(
+              t =>
+                gcsObjectUrisToValidate
+                  .traverse_(s => validateBucketObjectUri(userInfo.userEmail, t, s, context.traceId))
+            ) //TODO figure out implicits required for parTraverse
```

Comment:
I think `Parallel`

Reply:
yea, what's the import tho?

Reply:
`import cats.Parallel` is what I used in https://github.com/DataBiosphere/leonardo/pull/1309/files#diff-2b3a7f343790bfa0bfab5213c8dcfb90

Yeah it's weird it's not cats-effect..